core/Date.php

```diff
@@ -611,6 +612,38 @@ public function subYear($n)
         return new Date($result, $this->timezone);
     }
 
+    protected static $use12HourClock;
+
+    protected function getTimeFormat()
+    {
+        if (is_null(self::$use12HourClock)) {
+            self::$use12HourClock = LanguagesManager::uses12HourClockForCurrentUser();
```

Comment:
Ideally, core would not know about anything implemented by plugins and not access any code from plugins directly. I know it's sometimes not really possible though or it is possible but doesn't make it much better. @diosmosis should we maybe use DI in such a case? Like by default core would always return `true` for `date.uses12HourClockForCurrentUser` DI setting (`global.php`) and LanguagesManager plugin would overwrite it?